zkevm-circuits/src/evm_circuit/execution/error_oog_call.rs

```diff
@@ -240,27 +314,35 @@ mod test {
         rd_length: u64,
     }
 
-    fn caller(stack: Stack, caller_is_success: bool) -> Account {
-        let terminator = if caller_is_success {
-            OpcodeId::RETURN
-        } else {
-            OpcodeId::REVERT
-        };
-
-        // Call twice for testing both cold and warm access
-        let bytecode = bytecode! {
+    fn call_bytecode(
+        opcode: OpcodeId,
+        address: Address,
+        stack: Stack,
+        is_success: bool,
+    ) -> Bytecode {
+        let mut bytecode = bytecode! {
             PUSH32(Word::from(stack.rd_length))
             PUSH32(Word::from(stack.rd_offset))
             PUSH32(Word::from(stack.cd_length))
             PUSH32(Word::from(stack.cd_offset))
-            PUSH32(stack.value)
-            PUSH32(Address::repeat_byte(0xff).to_word())
+        };
+        if opcode == OpcodeId::CALL || opcode == OpcodeId::CALLCODE {
+            bytecode.push(32, stack.value);
+        }
+        bytecode.append(&bytecode! {
+            PUSH32(address.to_word())
             PUSH32(Word::from(stack.gas))
-            CALL
+            .write_op(opcode)
             PUSH1(0)
             PUSH1(0)
-            .write_op(terminator)
-        };
+            .write_op(if is_success { OpcodeId::RETURN } else { OpcodeId::REVERT })
```

Comment:
I think don't need `is_success` check since it is an error case.

Reply:
Deleted `is_success` parameter and always use `OpcodeId::REVERT` in test cases of OOG call.
Fixed in https://github.com/scroll-tech/zkevm-circuits/pull/262/commits/a597c2c92b44dc45bf0e9760914243fb91742259